### PR TITLE
Promoting the simple TCP server to be run from the command line

### PR DESCRIPTION
**Pull request by chadell**
* Add a simple TCP server in tcpgoon CLI, with basic input parameters (port, parallel connections, etc)
* Add simple end to end tests using tcpgoon client and validating the expected result

### CODE REVIEW

**Comment by chadell** [comment]
CI in Travis is failing on the new end-to-end test but the local `./_script/cibuild-docker` works fine. still trying to figure it out.

File: cmd/server.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+func runTcpgoonServer(params TCPServerParams) {
+	fmt.Println("Running the simple TCP server in port", params.port, "up to", params.maxconnections, "connections or", params.duration, "seconds, what happens first")
+
+	dispatcher := &tcpserver.Dispatcher{
+		Handlers: make(map[string]*tcpserver.Handler),
+		Lock:     sync.RWMutex{},
+	}
+
+	var end_waiter sync.WaitGroup
+	end_waiter.Add(1)
```
Comment (dcaba) [inline cmd/server.go:71]: avoid the usage of "_" in var and function names, not go friendly

File: cmd/server.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		}
+	}
+	go runTCPServer()
+
+	WaitForCtrlC(&end_waiter)
+
+	end_waiter.Wait()
+}
+
+func WaitForCtrlC(end_waiter *sync.WaitGroup) {
```
Comment (dcaba) [inline cmd/server.go:87]: with the capital you are marking this as public, and probably you dont want

**Review by dcaba — APPROVED** [review]
once this is amended with the comments, feel free to merge!

**Pull request merged**